### DISCUSSION TITLE
Convergence Issues in Parallel Field Inversion with Periodic Boundary Conditions

### DISCUSSION THREAD

**Discussion by lnhang0618** [mdolab/dafoam#611] [General]
Dear Professor He,

In our previous discussion, there was mention of convergence issues with the periodic boundary conditions in Field Inversion during parallelization. I experimented with version 3.11 and it seems that the problem still exists. What is the main cause of this issue?"

**Comment by friedenhe** [comment]
If I remember correctly, the problem is in the accuracy of the adjoint derivatives. The flow should still converge. We are still not quite sure what caused the issue...

**Reply by lnhang0618** [reply]
Yes, the adjoint derivatives are not correct when parallel, which causes issues in my attempt to directly train the neural network using dJdbeta.

**Comment by friedenhe** [comment]
Please run the periodic hill case in serial.

**Reply by uttamcadambi07** [reply]
What if we wanted to run another case that has periodic boundary conditions that is not the periodic hill case? For example we are interested in doing infinite span wing simulations and want to initialise the spanwise boundaries as periodic. It is impossible to run it in serial (we have some 2million cells) so can we still run it successfully in parallel? please advice

**Comment by friedenhe** [comment]
DAFoam's adjoint accuracy is good in parallel for other cases with periodic boundary conditions (e.g., propellers), it just somehow does not work for the periodic hill case. I suggest you run your case in parallel and test its accuracy. I may work without a problem, so there is no need to worry about it.

**Reply by 15175829310** [reply]
Is there a wedge boundary condition in dafoam?

**Comment by friedenhe** [comment]
@15175829310 no...

**Reply by 15175829310** [reply]
Thank you for your reply, but I need to perform flow field inversion on an axially symmetric shape. How should I do it if there is no "wedge" boundary condition? Looking forward to your reply.

**Comment by friedenhe** [comment]
I belive you can do it with the `cyclic` boundary condition. Check the Rotor37 tutorial.
